### PR TITLE
Use a new KAGGLE_GRPC_DATA_PROXY_URL env variable for gRPC proxying

### PR DESCRIPTION
**Pull request by Philmod**
http://b/308644984

### CODE REVIEW

File: patches/sitecustomize.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def new_configure(*args, **kwargs):
             client_options = kwargs['client_options']
         else:
             client_options = {}
-        client_options['api_endpoint'] = os.environ['KAGGLE_DATA_PROXY_URL']
+        client_options['api_endpoint'] = os.environ['KAGGLE_GRPC_DATA_PROXY_URL']
```
Comment (psbang) [inline patches/sitecustomize.py:104]: If transport is REST, wouldn't we want to use the old `KAGGLE_DATA_PROXY_URL`?

Comment (Philmod) [reply patches/sitecustomize.py:104]: Very good catch, thanks!

File: patches/sitecustomize.py [hunk 2]
```diff
@@ -101,10 +101,11 @@ def new_configure(*args, **kwargs):
...
         if os.getenv('KAGGLE_GOOGLE_GENERATIVE_AI_USE_REST_ONLY') != None:
             client_options['api_endpoint'] += '/palmapi'
             kwargs['transport'] = 'rest'
+            client_options['api_endpoint'] = os.environ['KAGGLE_DATA_PROXY_URL']
```
Comment (psbang) [inline patches/sitecustomize.py:108]: We would want this after line 111 as well, because that is also specifying a REST transport

Comment (psbang) [reply patches/sitecustomize.py]: @Philmod I applied the changes from this comment into the branch

Comment (Philmod) [reply patches/sitecustomize.py:110]: Thanks, I simplified a bit the code.

File: tests/test_google_generativeai_patch_disabled.py [hunk 3]
```diff
@@ -29,7 +30,6 @@ def test_disabled(self):
         env.set("KAGGLE_DATA_PROXY_TOKEN", "foobar")
         env.set("KAGGLE_DATA_PROXY_URL", self.http_endpoint)
         env.set("KAGGLE_GRPC_DATA_PROXY_URL", self.grpc_endpoint)
-        env.set("KAGGLE_GOOGLE_GENERATIVE_AI_USE_REST_ONLY", "True")
```
[the remark this answers is not in the document]
Comment (Philmod) [reply tests/test_google_generativeai_patch_disabled.py]: That was the goal: this test file is named `*_disabled`, I wanted to check it wasn't hitting the http server.

[the remark this answers is not in the document]
Comment (Philmod) [reply tests/test_google_generativeai_patch_disabled.py]: @psbang Unless I'm missing something?

**Pull request merged**